admix/uploader.py

```diff
@@ -85,6 +85,8 @@ def upload(path, rse,
             db.update_data(number, data_dict)
         try:
             clients.upload_client.upload(to_upload)
+            if not get_rule(did,rse):
```

Comment:
I actually have a question here. By the time this line is executed, has the upload processing the previous line finished? By finish, I mean that you will have a rucio rule added, with files inside it. If not, let's say the upload is still happening in the backend, will the `add_replication_rule` interfere with the process?

Reply:
No, it should not. As soon as the line above (clients.upload_client.upload) ends, the rule should be already created because the rule is created directly by the chain of functions called by that function. Said otherwise, the creation of the rule is not done by some Rucio daemon that requires time to do its job, but it's done directly by that client command.